pkg/util/util.go

```diff
@@ -185,3 +186,57 @@ func CanBeNonRoot(vmi *v1.VirtualMachineInstance) error {
 func MarkAsNonroot(vmi *v1.VirtualMachineInstance) {
 	vmi.Status.RuntimeUser = 107
 }
+
+func SetDefaultNetworkInterface(vmi *v1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig) error {
+	autoAttach := vmi.Spec.Domain.Devices.AutoattachPodInterface
+	if autoAttach != nil && *autoAttach == false {
+		return nil
+	}
+
+	// Override only when nothing is specified
+	if len(vmi.Spec.Networks) == 0 && len(vmi.Spec.Domain.Devices.Interfaces) == 0 {
+		iface := v1.NetworkInterfaceType(clusterConfig.GetDefaultNetworkInterface())
+		switch iface {
+		case v1.BridgeInterface:
+			if !clusterConfig.IsBridgeInterfaceOnPodNetworkEnabled() {
+				return fmt.Errorf("Bridge interface is not enabled in kubevirt-config")
+			}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		case v1.MasqueradeInterface:
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		case v1.SlirpInterface:
+			if !clusterConfig.IsSlirpInterfaceEnabled() {
+				return fmt.Errorf("Slirp interface is not enabled in kubevirt-config")
+			}
+			defaultIface := v1.DefaultSlirpNetworkInterface()
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*defaultIface}
+		}
+
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+	}
+	return nil
+}
```

Comment:
I think this code belongs on some network related package. Or as part of this `clusterconfig` pkg.